undofx/src/main/java/org/fxmisc/undo/impl/UndoManagerImpl.java

```diff
@@ -209,7 +229,7 @@ private void addChange(C change) {
     @SuppressWarnings("unchecked")
     private C[] merge(C c1, C c2) {
         Optional<C> merged = merge.apply(c1, c2);
-        if(merged.isPresent()) {
+        if(merged.isPresent() && isNotIdentity.test(merged.get())) {
             return (C[]) new Object[] { merged.get() };
         } else {
             return (C[]) new Object[] { c1, c2 };
```

Comment:
Yes, that's what I intended.

Taking RichTextFX as an example, say you type in `hello` and then hit backspace 5 times. With my version, the undo history will now be empty. With yours, it will contain 2 items:
 - at 0, insert `h`
 - at 0, delete `h`

I would say yours is as far from what actually happened as mine.

Plus, why merge the  first 4 deletions and not the 5th one? Seems somewhat irregular and arbitrary.

Plus, if the user didn't want changes to annihilate, they could have provided `merge` that would return `Optional.empty()` instead of an identity change. Thus this behavior could be implemented without any new support from UndoFX.